dialogue-core/src/main/java/com/palantir/dialogue/core/QueuedChannel.java

```diff
@@ -88,17 +74,6 @@
     QueuedChannel(LimitedChannel delegate, int maxQueueSize, DispatcherMetrics metrics) {
         this.delegate = delegate;
         this.queuedCalls = new LinkedBlockingDeque<>(maxQueueSize);
-        this.backgroundScheduler.scheduleWithFixedDelay(
```

Comment:
Without this you can end up with requests that are queued forever:

1. Submit 2 requests, but only have capacity for 1
2. Request 1 finishes and tries to schedule next request but still no capacity
3. Request 2 is stuck

Reply:
Forever and 100ms are roughly equivalent here. See the comments on signaling availability for a better long-term solution.

Reply:
I'm not sure I understand how "forever" and "100ms" are roughly equivalent?

As far as having LimitedChannels signal that they can accept requests: I'd have to see a proposal for how this works. It sounds like the sort of thing that could get a bit hairy.